scripts/shared/clusters.sh

```diff
@@ -114,7 +114,57 @@ function deploy_cni() {
 
 function deploy_weave_cni(){
     echo "Applying weave network..."
-    curl -sL "https://cloud.weave.works/k8s/net?k8s-version=v$k8s_version&env.IPALLOC_RANGE=${cluster_CIDRs[${cluster}]}" | sed 's!ghcr.io/weaveworks/launcher!weaveworks!' | kubectl apply -f -
+
+    WEAVE_YAML=`curl -sL "https://cloud.weave.works/k8s/net?k8s-version=v$k8s_version&env.IPALLOC_RANGE=${cluster_CIDRs[${cluster}]}" | sed 's!ghcr.io/weaveworks/launcher!weaveworks!'`
+
+    # Search the YAML for images that need to be downloaded
+    IMAGE_LIST=( $(echo "${WEAVE_YAML}" | grep "image:" | awk -F"'" '{print $2}') )
+    echo "IMAGE_LIST=${IMAGE_LIST[@]}"
+    for i in "${!IMAGE_LIST[@]}"
```

Comment:
We don’t use the index, so we can iterate over the array contents directly:

```suggestion
    for image in "${IMAGE_LIST[@]}"
```

Reply:
Done.